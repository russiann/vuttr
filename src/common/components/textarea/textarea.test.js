import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Textarea from './textarea.component';

afterEach(cleanup);

it('renders', () => {
  const {asFragment} = render(<Textarea />);
  expect(asFragment()).toMatchSnapshot();
});

it('should render default value', () => {
  const text = 'Hello World!';
  const {getByTestId} = render(<Textarea defaultValue={text} />);
  const input = getByTestId('textarea');
  expect(input.value).toBe(text);
});

it('should render typed text', () => {
  const text = 'Hello World!';
  const {getByTestId} = render(<Textarea />);
  const input = getByTestId('textarea');
  fireEvent.change(input, {target: {value: text}});
  expect(input.value).toBe(text);
});
