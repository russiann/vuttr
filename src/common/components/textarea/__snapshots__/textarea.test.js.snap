// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <textarea
    class="css-qpvf45-TextArea egmejh80"
    data-testid="textarea"
    type="text"
  />
</DocumentFragment>
`;

exports[`snapshot 1`] = `
<DocumentFragment>
  <textarea
    class="css-qpvf45-TextArea egmejh80"
    data-testid="textarea"
    type="text"
  />
</DocumentFragment>
`;
